Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -0,0 +1,293 @@
+
+function getPublishProfileUtil ( SPN, webAppName: string, resourceGroupName: string, publishMethod: string, deployToSlotFlag: boolean, slotName: string ): Q.Promise<any> {
+
+    var deferred = Q.defer();
+    var insertInUrl = deployToSlotFlag ? "/slots/" + slotName : "";
+    getAuthorizationToken(SPN)
+        .then(function (accessToken) {
+        var requestOptions = {
+            url: 'https://management.azure.com/subscriptions/' + SPN.subscriptionId + '/resourceGroups/' + resourceGroupName + '/providers/Microsoft.Web/sites/' + webAppName + insertInUrl + '/publishxml?api-version=2015-08-01',
+            auth: {
+                bearer: accessToken
+            },
+            method: 'POST'
+        };
+        request(requestOptions, function (error, response, body) {
+            if (error)
+                deferred.reject(error);
+            else {
+                if (response.statusCode === 200) {
+                    parseString(body, function (error, result) {
+                        for (var i in result.publishData.publishProfile) {
+                            if (result.publishData.publishProfile[i].$.publishMethod === publishMethod)
+                                deferred.resolve(result.publishData.publishProfile[i].$);
+						}
+                        deferred.reject('Error : No Such Deploying Method Exists');
+                    });
+                }
+                else {
+                    deferred.reject('Error Fetching Deployment Publish Profile [Staus Code : '+response.statusCode+']');
+                }
+            }
+        });
+    }, function (error) {
+        taskLib.error(error);
+    });
+
+    return deferred.promise;
+}
+exports.getAzureRMWebAppPublishingProfileDetails = function (SPN, webAppName: string, resourceGroupName: string, publishMethod: string, deployToSlotFlag: boolean, slotName: string): Q.Promise<any> {
+    var deferred = Q.defer();
+      
+    if ( !deployToSlotFlag ) {
+        getAzureRMWebAppDetails_version2 ( SPN, webAppName, 'Microsoft.Web/sites' ).then ( function (webApp) {
+            resourceGroupName = webApp.id.split ('/')[4];
+            getPublishProfileUtil ( SPN, webAppName, resourceGroupName, publishMethod, deployToSlotFlag, slotName).then ( function (publishProfile) {
+                deferred.resolve (publishProfile);
+            }, function (error) {
+                deferred.reject (error);
+            });
+        }, function ( error ) {
+            deferred.reject (error);
+        });
+    } 
+    else  {
+        
+        getPublishProfileUtil ( SPN, webAppName, resourceGroupName, publishMethod, deployToSlotFlag, slotName).then ( function (publishProfile) {
+                deferred.resolve (publishProfile);
+        }, function (error) {
+            deferred.reject (error);
+        });
+
+    }
+    
+    return deferred.promise;
+};
+
+function getAzureRMWebAppDetails_version2 ( SPN, webAppName : string, resourceType : string ) : Q.Promise<any> {
```

Comment:
This method is returning wrong webapp details , since in response getting more than on basis of sub-string and selecting only first one.